core/domain/suggestion_services_test.py

```diff
@@ -450,6 +450,88 @@ def test_check_can_resubmit_suggestion(self):
             self.suggestion_id, self.normal_user_id)
         self.assertEqual(can_resubmit, False)
 
+    def test_edit_suggestion(self):
+        with self.swap(
+            feedback_models.GeneralFeedbackThreadModel,
+            'generate_new_thread_id', self.mock_generate_new_thread_id):
+            with self.swap(
+                exp_services, 'get_exploration_by_id',
+                self.mock_get_exploration_by_id):
+                suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id, self.target_version_at_submission,
+                    self.author_id, self.change, 'test description',
+                    self.reviewer_id)
+        new_value = {
+            'html': 'edited suggestion'
+        }
+        new_change_dict = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'state_name': 'state_1',
+            'new_value': new_value,
+            'old_value': 'new suggestion content'
+        }
+
+        suggestion = (suggestion_services.get_suggestion_by_id(
+            self.suggestion_id))
+        change_cls = type(suggestion.change)
+        change_object = change_cls(new_change_dict)
+        suggestion_services.edit_suggestion(suggestion, change_object)
+
+        suggestion = (suggestion_services.get_suggestion_by_id(
+            self.suggestion_id))
+        suggestion_change_dict = suggestion.change.to_dict()
+        self.assertEqual(new_change_dict, suggestion_change_dict)
+
+    def test_can_edit_suggestion(self):
```

Comment:
Is it resolved?

Reply:
I have written it in this function. You can see it
Thanks

Reply:
I'm not sure when you have written this, but in general, try adding a reply once you resolve a comment.

Reply:
Ok

>     * Invalid activity type.

While related to this I think it is checked while creating suggestion because in edit suggestion API there is only submitting of suggestion id and change object so there is no need of checking this.

>     * Suggestion doesn't exist.

Its implemented you can check it 

>     * Suggestions related to all different type of activity.

currently this PR focus on suggestion related to exploration. So we were not focussing on different other suggestion types. We have discussed also @nithusha21.